syntax = "proto3";

package pkda;

message RegisterClientRequest {
    string client_id = 1;
    string client_public_key = 2;
    string client_address = 3;
    int64 timestamp = 4;
}

message RegisterClientResponse {
    string pkda_public_key = 2;
    int64 timestamp = 4;
}

message PublicKeyRequest {
    string client_id = 1;
    int64 timestamp = 2;
}

message EncryptedMessage {
    string encrypted_message = 1;
}

service PKDA {
    rpc RegisterClient(RegisterClientRequest) returns (RegisterClientResponse);
    rpc GetPublicKey(PublicKeyRequest) returns (EncryptedMessage);
}

service Client {
    rpc SendMessage(EncryptedMessage) returns (EncryptedMessage);
}
